utils/src/main/java/com/nedap/archie/query/APathQuery.java

```diff
@@ -34,18 +31,25 @@
     private List<PathSegment> pathSegments = new ArrayList<>();
 
     public APathQuery(String query) {
-        if(!query.equals("/")) {
+        if(!query.startsWith("/") && !query.contains("/") && !query.contains("[")) {
+            pathSegments.add(new PathSegment(query));
+        } else if(!query.equals("/")) {
             XPathLexer lexer = new XPathLexer(new ANTLRInputStream(query));
             XPathParser parser = new XPathParser(new CommonTokenStream(lexer));
             LocationPathContext locationPathContext = parser.locationPath();
             AbsoluteLocationPathNorootContext absoluteLocationPathNorootContext = locationPathContext.absoluteLocationPathNoroot();
-            if (absoluteLocationPathNorootContext == null) {
-                throw new UnsupportedOperationException("relative xpath expressions not yet supported: " + query);
-            }
-            if (!absoluteLocationPathNorootContext.getTokens(XPathLexer.ABRPATH).isEmpty()) {
+            //if (absoluteLocationPathNorootContext == null) {
+            //    throw new UnsupportedOperationException("relative xpath expressions not yet supported: " + query);
+           // }
```

Comment:
Are relative xpath expressions now supported? Then these lines can be removed completely. If not, maybe re-add the exception?

Reply:
I think all our path queries are treated as relative to the context node (so, c_object or rm object), instead of from the document root. So I would say we have only supported relative paths so far, even if all paths had to start with a '/' before this change.
I think this change is mainly so that path lookup in bmm is a tiny bit easier to implement. I don't think it's possible to introduce the relative/absolute distinction easily, and I'm not sure if it matters much.

Reply:
This only matters inside a) actual RM structures, e.g. the path functions LOCATABLE can be called on any node in an instance structure and b) actual archetypes, e.g. relative paths in differential archetypes. Not sure if that is what is being handled here.

Reply:
if we follow the XPath rules, this matters whenever you call a query on objects that are not document roots. So if you do CObject.itemAtPath, or use AOMPathQuery, or RMPathQuery but not on the archetype root or at the RM document root, often a composition. 
Currently we handle everything as if they are relative queries, so the tree below the given object is searched.
If we would follow xpath, then if it starts with a '/' it is an absolute query, and lookup must start at the document root. if it starts without a '/' at the start, it must start at the given context node instead

However in differential archetypes, these paths in ADL always start with a slash, and they are currently directly passed to this query. So these rules are not always followed in the specification.

For BMM, absolute paths make no sense since there is no concept of a document root in these queries, since they do not use packages.

So what it does in the current master-branch:
- it accepts only paths starting with a '/'
- these are being treated as relative queries

What it does after this change:
- it accepts paths starting with either '/' or without one.
- these are all being treated as relative queries

I say this is sort of an improvement? 
If we ever need to fix this, it is possible, but it is a major change because we need to change it everywhere these lookups are used, and need to communicate this very clearly to users, or perhaps just copy the implementation, deprecate the old one and allow ourselves and others to gradually migrate.

The one lookup that does it properly is the RMQueryContext - because that translates these queries to xpath and does an actual XPath on the XML, using the JAXB XML binding to return RM Objects instead of XML nodes. But that does not use this parser convert this, it uses the ANTLR grammar directly. In Archie that is only used in the rule evaluation and a very limited number of tests.